### PR TITLE
Avoid fetching traces when the search criteria is not changed

### PR DESCRIPTION
**Pull request by tacigar**
In the previous implementation, fetch processing was performed each time the search page was opened, even if the search criteria were the same.
This implementation degrades UX significantly because fetch processing is slow.
We should avoid fetching traces when the search criteria is not changed.
This PR do so.

Related: #2685 
If we do not avoid unnecessary fetching, the page transition of `search page` → `see details` → `back button` → `search page` UX will be the worst....

### CODE REVIEW

File: zipkin-lens/src/actions/traces-action.test.js [hunk 1]
```diff
@@ -85,6 +76,10 @@ describe('traces async actions', () => {
         traces: rawTraces,
         correctedTraceMap,
         traceSummaries,
+        lastFetchingParams: {
```
Comment (codefromthecrypt) [inline zipkin-lens/src/actions/traces-action.test.js]: maybe lastQueryParams? if the relates to the query api (/api/v2/traces) https://zipkin.io/zipkin-api/#/default/get_traces

**Review by codefromthecrypt — COMMENTED** [review]
only thing curious about is if a threshold is useful. Ex if 5 or 10 seconds later, it will do a refresh when clicking back. Maybe that isn't useful? not sure.

**Comment by tacigar** [comment]
I think a threshold is not useful...
If the user want to refresh search results, the user only push Enter key.

**Comment by tacigar** [comment]
PTAL 🙇

**Comment by codefromthecrypt** [comment]
agree about press enter is intuitive. thanks for thinking about it!

**Pull request merged**